allennlp/common/params.py

```diff
@@ -354,6 +354,49 @@ def from_file(params_file: str, params_overrides: str = "") -> 'Params':
 
         return Params(param_dict)
 
+    def to_file(self, params_file: str, preference_orders: List[List[str]] = None):
+        with open(params_file, "w") as handle:
+            json.dump(self.as_ordered_dict(preference_orders), handle, indent=4)
+
+    def as_ordered_dict(self, preference_orders: List[List[str]] = None):
+        """
+        Returns Ordered Dict of Params from list of partial order preferences.
+
+        Parameters
+        ----------
+        preference_orders: List[List[str]], optional
+            ``preference_orders`` is list of partial preference orders. ["A", "B", "C"] means
+            "A" > "B" > "C". For multiple preference_orders first will be considered first.
+            Keys not found, will have last but alphabetical preference. Default Preferences:
+            ``[["dataset_reader", "iterator", "model", "train_data_path", "validation_data_path",
+                "test_data_path", "trainer", "vocabulary", "seeds"], ["type"]]``
+        """
+        params_dict = self.as_dict(quiet=True)
+        if not preference_orders:
+            preference_orders = []
+            preference_orders.append(["dataset_reader", "iterator", "model",
+                                      "train_data_path", "validation_data_path", "test_data_path",
+                                      "trainer", "vocabulary", "seeds"])
+            preference_orders.append(["type"])
+
+        def preference_ranks(key):
+            '''returns list of all preference scores'''
+            return [order.index(key) if key in order else len(order)
+                    for order in preference_orders]
+
+        def order_func(key):
+            '''returns list of all preference scores with default alphabetic score'''
+            return preference_ranks(key) + [key]
+
+        def _order_dict(dictionary, order_func):
```

Comment:
This is already defined inside of a function; no need to prepend an underscore to the name.

Reply:
Sure 👍

Reply:
done 👍